_posts/2022-01-28-filesystem-encryption-integrity.adoc

```diff
@@ -0,0 +1,145 @@
+please refer to the https://docs.wildfly.org/20/WildFly_Elytron_Security.html[Elytron documentation].
+
+We then configure a simple role decoder and create a new security domain which will make use of our
+filesystem realm and role decoder as follows:
+[source]
+----
+/subsystem=elytron/simple-role-decoder=from-roles-attribute:add(attribute=Roles)
+/subsystem=elytron/security-domain=fsDomain:add(realms=[{realm=fsRealm, role-decoder=from-roles-attribute}], default-realm=fsRealm,permission-mapper=default-permission-mapper)
+----
+
+
+NOTE: Creating an additional security domain (``fsDomain``in this case) is not necessary.
+We could alternatively take the default ``ApplicationDomain`` and add the
+FileSystem realm and role-decoder to it.
+
+We then add our security domain mapping to the Undertow subsystem, update the ``sasl-authentication-factory`` to use
+our security domain, add the ``PLAIN`` authentication mechanism and update the ``http-remoting-connector`` to use our ``sasl-authentication-factory``:
+
+
+[source]
+----
+/subsystem=undertow/application-security-domain=other:write-attribute(name=security-domain, value=fsDomain)
+/subsystem=elytron/sasl-authentication-factory=application-sasl-authentication:write-attribute(name=security-domain, value=fsDomain)
+/subsystem=elytron/sasl-authentication-factory=application-sasl-authentication:list-add(name=mechanism-configurations, value={mechanism-name=PLAIN})
+/subsystem=remoting/http-connector=http-remoting-connector:write-attribute(name=sasl-authentication-factory,value=application-sasl-authentication)
+----
+
+== Configuring the Client (wildfly-config.xml)
+
+The username and credentials to establish the connection to the application server are configured in the
+``wildfly-config.xml`` file.
+
+[source,xml]
+----
+ <authentication-configurations>
+            <configuration name="default-config">
+                <set-user-name name="quickstartUser"/>
+                <credentials>
+                    <clear-password password="password123!"/>
+                </credentials>
+                <sasl-mechanism-selector selector="PLAIN"/>
+                ...
+            </configuration>
+</authentication-configurations>
+----
+
+== Verifying Encryption and Integrity
+
+From the root directory of the *https://github.com/wildfly-security-incubator/elytron-examples/blob/master/encryption-integrity-filesystem-realm/[quickstart example]* run the following command the deploy the web application to wildfly
+[source]
+----
+mvn clean install wildfly:deploy
+----
+
+Now you may navigate to ``http://localhost:8080/encryption-integrity-filesystem``, and when it prompt's you to enter a username and password, put in the credentials we specified earlier, ``quickstartUser``, and ``password123!``. This should authenticate you to a page that shows you the principal you're logged in with.
+
+This login process demonstrates that the encryption and integrity is working correctly.
```

Comment:
Maybe this could be reworded to something like:

The successful login indicates that things have been configured correctly.

(Or something like that.)

Reply:
Updated